site/app/controllers/HomePageController.php

```diff
@@ -135,21 +136,68 @@ function(Course $course) {
      * @return Response
      */
     public function showHomepage() {
-        $courses = $this->getCourses()->json_response->json;
+        $courses = $this->getCourses()->json_response->json;        
+        $archived_courses = $courses["data"]["archived_courses"];
+        $unarchived_courses = $courses["data"]["unarchived_courses"];
+        $course_types = [$unarchived_courses, $archived_courses];
+        $rank_titles = [
+            User::GROUP_INSTRUCTOR              => "Instructor:",
+            User::GROUP_FULL_ACCESS_GRADER      => "Full Access Grader:",
+            User::GROUP_LIMITED_ACCESS_GRADER   => "Grader:",
+            User::GROUP_STUDENT                 => "Student:"
+        ];
+        $statuses = array();
+
+        foreach($course_types as $course_type) {
+
+            $ranks = array();
+
+            //Create rank lists
+            for ($i = 1; $i < 5; $i++){
+                $ranks[$i] = [];
+                $ranks[$i]["title"] = $rank_titles[$i];
+                $ranks[$i]["courses"] = [];
+            }
+
+            foreach ($course_type as $course) {
+                array_push($ranks[$course["user_group"]]["courses"], $course);
+            }
+
+            //Filter any ranks with no courses
+            $ranks = array_filter($ranks, function($rank) {
+                return count($rank["courses"]) > 0;
+            });
+            $statuses[] = $ranks;
+        }
```

Comment:
This logic can live in the View as it's purely related to setting up stuff for displaying.